src/commonMain/kotlin/com/algolia/search/transport/HttpRequestBuilder.kt

```diff
@@ -9,12 +9,12 @@ import io.ktor.client.request.header
 import io.ktor.client.request.parameter
 
 
-internal fun HttpRequestBuilder.setApplicationId(applicationID: ApplicationID) {
-    header(KeyAlgoliaApplicationID, applicationID.raw)
+internal fun HttpRequestBuilder.setApplicationId(applicationID: ApplicationID?) {
```

Comment:
Why allowing nullable here? Is there a legitimate case? :thinking:

Reply:
🤷‍♂